### ISSUE TITLE
Small improvements to Label

### ISSUE THREAD

**Issue by birdsarah** [bokeh/bokeh#4179]
The following, where 'color' is a column in your ColumnDataSource fails.

```python
    Label(
        x='x0', y='y0', text='name', level='glyph', text_color='color',
        x_offset=5, y_offset=5, source=source, render_mode='canvas')
    )
```

Would be cool if it didn't.

**Comment by birdsarah** [comment]
ping @canavandl

**Comment by birdsarah** [comment]
Also, I was tripped up by the following:

```python
p.add_annotation(
    Label(
        x=cluster_0_x_0, y=cluster_0_y_0, text=['Cluster0'])
)
```

I have to put text in a list, but not the x & y.

(This is ringing a bell as something I've complained about in the past and isn't unique to Label)

**Comment by birdsarah** [comment]
These are both ringing a bell as something I've complained about in the past and isn't unique to Label.

Leaving this here for now to see if there's an old issue already covering.

**Comment by mattpap** [comment]
The issue with text is that it uses `TextSpec` property descriptor, which allows fields (as string) and then you can't differentiate between a string field name and string value (scalar), so you have to go through a vector of values. This is a very old issue we have `Text` glyph. Surprisingly (to me) `Label` allows to set a data source, so this all makes sense. Well kind of. At this point I don't know what's the difference between `Text` glyph and `Label` annotation.

**Comment by bryevdv** [comment]
Label can render to css to support latex, can have a filled background, will probably define anchor points to attach to arrows, and can be positioned in screen coordinates (does not participate in hit testing)

**Comment by canavandl** [comment]
@mattpap is right about TextSpec and ColorSpec being unable to distinguish between a string that's intended to be a value or a field. I think it's a limitation of our current type system - so you're stuck with either enclosing the string in a list or specifying you're passing a value using `{"value": "red"}`.

**Comment by bryevdv** [comment]
There is also `value("red")` and `field("foo")` I agree it sucks, for glyphs I think we have to be completely consistent that strings are field names, that is the case in 98% of instances. I would personally at least entertain the idea of being more flexible or reversing this convention for the `Label` annotation specifically, but maybe that is also confusing.

**Comment by canavandl** [comment]
resolved by: #4309
